lib/logger_humio_backend.ex

```diff
@@ -191,51 +192,110 @@ defmodule Logger.Backend.Humio do
     IO.puts(io_device, [level, ": ", message])
   end
 
-  def default_config do
-    [
-      # used primarily for testing
-      client: Client.Tesla,
-      ingest_api: IngestApi.Structured,
-      iso8601_format_fun: TimeFormat.iso8601_format_fun(),
-      # "advertised" options
-      host: "",
-      token: "",
-      level: :debug,
-      metadata: [],
-      format: nil,
-      max_batch_size: 20,
-      flush_interval_ms: 2_000,
-      debug_io_device: :stdio,
-      fields: %{},
-      tags: %{}
-    ]
+  def default_config, do: @default_config
+
+  @spec configure(Keyword.t(), t()) :: t()
+  defp configure(opts, state) do
+    updates =
+      Application.get_all_env(:logger_humio_backend)
+      |> Keyword.merge(opts)
+
+    struct!(state, updates)
+    |> compile_format()
   end
 
-  defp configure(name, opts) do
-    env = Application.get_env(:logger, name, [])
-    opts = Keyword.merge(env, opts)
-    Application.put_env(:logger, name, opts)
+  defp compile_format(%__MODULE__{format: format} = state) when is_binary(format) do
+    %{state | format: Formatter.compile(format)}
+  end
 
-    config =
-      default_config()
-      |> Keyword.merge(opts)
-      |> Keyword.put_new(:name, name)
-      |> Keyword.update!(:format, fn format -> Formatter.compile(format) end)
-      |> Keyword.update!(:token, fn token -> token(token) end)
-      |> Enum.into(Map.new())
+  defp compile_format(state) do
+    state
+  end
+
+  defp transmit(
+         %__MODULE__{
+           host: host,
+           token: token,
+           client: client,
+           tags: tags
+         } = state
+       ) do
+    headers = generate_headers(token, @content_type)
+
+    to_events(state)
```

Comment:
It's conventional to put the first argument on its own line at the beginning of a pipeline.
```suggestion
    state
    |> to_events()
```

Reply:
On second thought, could these first two steps (`to_events`, `to_request`) be combined into one function? One of them takes the entire state and one takes a portion of it. If so, you could have something a little more concise like:

```elixir
case Jason.encode(to_request(state)) do
  #... 
```

Reply:
Makes sense.